paasta_tools/marathon_tools.py

```diff
@@ -993,3 +1015,47 @@ def kill_task(client, app_id, task_id, scale):
             return []
         else:
             raise
+
+
+def get_instances_from_zookeeper(service, instance):
+    with ZookeeperPool() as zookeeper_client:
+        (instances, _) = zookeeper_client.get('/autoscaling/%s/%s/instances' % (service, instance))
+        return int(instances)
+
+
+def update_instances_for_marathon_service(service, instance, delta):
+    zookeeper_path = '/autoscaling/%s/%s/instances' % (service, instance)
+    marathon_config = load_marathon_service_config(
+        service=service,
+        instance=instance,
+        cluster=load_system_paasta_config().get_cluster(),
+        load_deployments=False,
+    )
+    with ZookeeperPool() as zookeeper_client:
+        instances = min(
+            marathon_config.get_max_instances(),
+            max(marathon_config.get_min_instances(), marathon_config.get_instances() + delta))
+        zookeeper_client.ensure_path(zookeeper_path)
+        zookeeper_client.set(zookeeper_path, instances)
+        return instances
+
+
+class ZookeeperPool(object):
+    counter = 0
+    zk = None
+
+    @classmethod
+    def __enter__(cls):
+        if cls.zk is None:
+            cls.zk = KazooClient(hosts=load_system_paasta_config().get_zk_hosts(), read_only=True)
```

Comment:
I'm concerned here because this returns literally the zk hosts, and not the namespace we want.

I can't tell from the docs/source, but we should be able to pass in the correct "chroot" with our hosts  string.

https://kazoo.readthedocs.org/en/latest/_modules/kazoo/client.html#KazooClient.set_hosts

This is important long term as we don't want to make the assumption that we have the zk server all to ourselves and we pollute the root namespace.